### ISSUE TITLE
Inconsistent names in TTF Names table when using makegroups 4

### ISSUE THREAD

**Issue by sjml** [ryanoasis/nerd-fonts#1679]
### Requirements

- [X] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my issue and found nothing related or helpful
- [X] I have searched the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting) for help
- [X] I have checked the [Wiki](https://github.com/ryanoasis/nerd-fonts/wiki) for help

### Experienced Behavior

I'm trying to manually patch a font I created, but I'm getting inconsistent entries in the TTF Names table. When I open the resulting file in FontForge, I see two entries, one for US and one for British of the 'Preferred Family' entry, but they are different. 

<img width="631" alt="image" src="https://github.com/user-attachments/assets/2ed4f6ab-9acb-4ab1-908f-c3cb6bcf28b6">

I've traced this to the [`pfam_to_sfnt`](https://github.com/ryanoasis/nerd-fonts/blob/a2697b0fefe5e8d946c18a167a9496c6f224d7c9/bin/scripts/name_parser/FontnameParser.py#L312) function in the font name parser. It looks like the problem is that when return array is constructed (line 319) it's pulling the same index (`i`) from the `languages` and `names` array, but the layout of those arrays are not directly related; one is US/British and the other is long name / short name, so the resulting array has US: Long Name and British: Short Name. 

I would submit a PR directly to fix it, but I'm not sure what the desired behavior is... should only one set of names (either short long) be going in, or should the returned array be larger, using all permutations?

I only notice this because I'm trying to use `--makegroups 4` to get the "NF" shorter name, so perhaps it's an unsupported path OR I'm simply not using it correctly. (I am very open to the latter possibility.) 

Thanks for whatever help you can offer.

**Comment by Finii** [comment]
Thanks for reporting!

This is a duplicate of #1673

There are different possible names because the names can be different for different languanges (and even for different platforms), why does everyone expect that the names are identical for all languages? In that case different possible names per language does not make any sense ;-)

But the real reasons are given in 1673. I know this is ugly but it is a kludge to make Visual Studio happy :-(
As long as noone is hampered by it... ?

**Comment by sjml** [comment]
Ack, I'm so sorry I missed that previous bug in my searching! Nothing worse than dupes. 

I think the issue I was running into was two-fold -- there was an issue where some program was not finding it under "NF" but insisting I use "Nerd Font." Once I figured out the problem, I found the differing names in the file and assumed it was a but because it didn't seem to match the documented behavior of `makegroups 4`. I forget which program I was running into problems with, unfortunately. (Ironically, I think it may have been VS Code?) In any case, the different names are not simply display issues at least. 

It may just be a documentation issue -- maybe just add a note to the documentation that the Preferred Family will not be abbreviated when using `makegroups` >= 4? 

> why does everyone expect that the names are identical for all languages? In that case different possible names per language does not make any sense ;-)

I literally laughed at this. :-) Internationalization is a good thing, indeed! Just didn't expect "Nerd Font" to be something affected by a US/GB difference. Unless the Brits started spelling it "fount"...
